share/eds/store.go

```diff
@@ -70,24 +79,50 @@ func NewStore(basepath string, ds datastore.Batching) (*Store, error) {
 	}
 
 	return &Store{
-		basepath: basepath,
-		dgstr:    dagStore,
-		topIdx:   invertedRepo,
-		carIdx:   fsRepo,
-		mounts:   r,
+		basepath:   basepath,
+		dgstr:      dagStore,
+		topIdx:     invertedRepo,
+		carIdx:     fsRepo,
+		gcInterval: defaultGCInterval,
+		mounts:     r,
 	}, nil
 }
 
-// Start starts the underlying DAGStore.
 func (s *Store) Start(ctx context.Context) error {
+	s.ctx, s.cancel = context.WithCancel(context.Background())
+	go s.gc(s.ctx)
+
 	return s.dgstr.Start(ctx)
```

Comment:
I know you're taking care of this in another PR but just make sure you pass the long-lived ctx

Reply:
done